### PR TITLE
Auto complete input

### PR DESCRIPTION
**Pull request by phoskee**
Correctly created the PR (I guess) and fixed the bug discussed on #818

### CODE REVIEW

File: nicegui/elements/input.py [hunk 1]
```diff
@@ -52,6 +54,25 @@ def toggle_type(_):
 
         self.validation = validation
 
+        if autocomplete is not None:
+            def AutoCompleteInput():
+                if len(self.value) > 1:
```
Comment (rodja) [inline nicegui/elements/input.py:59]: Why do you only start autocompletion after the second character typed in?

**Review by rodja — CHANGES_REQUESTED** [review]
That is super cool. Could you also add a demo to [label_documentation.py](https://github.com/zauberzeug/nicegui/blob/main/website/more_documentation/label_documentation.py)? See [image_documentation.py](https://github.com/zauberzeug/nicegui/blob/main/website/more_documentation/image_documentation.py) for an example on how additional demos are placed on the documentation page.

File: nicegui/elements/input.py (hunk 1 above)
Comment (rodja) [reply nicegui/elements/input.py:59]: Ok, without `if` is a bit too much. But I think `> 0` would be more intuitive. Take vscode autocompletion for example. It starts suggesting as soon as you start typing.

**Comment by rodja** [comment]
> You meant input_documentation.py ?

Yes. Sorry for the mix up.

**Review by rodja — APPROVED** [review]

**Comment by rodja** [comment]
Wonderful! Thank you so much for adding this.

**Comment by falkoschindler** [comment]
I just reviewed and improved the code a little bit.

- To support Python 3.7, we need to use `List` instead of `list` for type annotations.
- When typing `tab` without any valid match, the input was reset to value `None`. I fixed it so that the current value is preserved in this case.

I'm still unsure about the `keydown.tab` event. Using `keydown.tab.prevent` feels a bit more natural (the focus stays on the input element), but prevents the user from tabbing out of the element to focus the next one.

**Comment by rodja** [comment]
Ah yes. Maybe the merge was a bit early. Sorry about that. Could we maybe use `keydown.tab.prevent` and  jump to the next element if we do not have something to autocomplete?

**Pull request merged**